// notion-crawler/dist/module.js
import {getAllPagesInSpace as $6ueHU$getAllPagesInSpace, uuidToId as $6ueHU$uuidToId, getCanonicalPageId as $6ueHU$getCanonicalPageId, getBlockTitle as $6ueHU$getBlockTitle, getTextContent as $6ueHU$getTextContent, normalizeTitle as $6ueHU$normalizeTitle} from "notion-utils";
import {NotionAPI as $6ueHU$NotionAPI} from "notion-client";



const $6592d0df9d7b02dd$var$notion = new (0, $6ueHU$NotionAPI)();
async function $6592d0df9d7b02dd$export$2e2bcd8739ae039(rootNotionPageId, spaceId, config = {}) {
    const pageMap = await (0, $6ueHU$getAllPagesInSpace)(rootNotionPageId, spaceId, $6592d0df9d7b02dd$var$notion.getPage.bind($6592d0df9d7b02dd$var$notion), {
        traverseCollections: true,
        uuid: true,
        ...config
    });
    const pageBlocks = Object.keys(pageMap).reduce((map, pageId)=>{
        const notionPageId = (0, $6ueHU$uuidToId)(pageId);
        const recordMap = pageMap[pageId];
        if (!recordMap) throw new Error(`Error loading page "${pageId}"`);
        const canonicalPageId = (rootNotionPageId || "").split("-").join("") === (pageId || "").split("-").join("") ? "/" : (0, $6ueHU$getCanonicalPageId)(pageId, recordMap, {
            uuid: false
        });
        const block = pageMap[pageId].block[pageId];
        const blockValue = block.value;
        const title = (0, $6ueHU$getBlockTitle)(blockValue, recordMap);
        const response = {
            ...map,
            [notionPageId]: {
                id: notionPageId,
                parentId: blockValue.parent_id.split("-").join(""),
                canonicalPageId: canonicalPageId,
                title: title,
                rawData: blockValue
            }
        };
        if (blockValue.parent_table === "collection") {
            const collection = pageMap[pageId].collection[blockValue.parent_id];
            if (collection?.value?.name) {
                const collectionName = (0, $6ueHU$getTextContent)(collection.value.name);
                const canonicalPageId = (0, $6ueHU$normalizeTitle)(collectionName);
                response[notionPageId].collection = {
                    name: collectionName,
                    canonicalPageId: canonicalPageId,
                    rawData: collection
                };
                response[notionPageId].canonicalPageId = `${canonicalPageId}/${response[notionPageId].canonicalPageId}`;
            }
        }
        return response;
    }, {});
    const appendParentSlug = (pageId, previous = "")=>{
        const page = pageBlocks[pageId];
        if (!page?.parentId || !pageBlocks[page.parentId]) return previous;
        return appendParentSlug(page.parentId, `${pageBlocks[page.parentId].canonicalPageId}/${previous}`);
    };
    const notionPageIdToSlugMapper = {};
    for(const key in pageBlocks){
        pageBlocks[key].slug = `${appendParentSlug(key, pageBlocks[key].canonicalPageId)}`;
        // TODO: Temp fix. Refactor this.
        if (pageBlocks[key].slug.startsWith("//")) pageBlocks[key].slug = pageBlocks[key].slug.slice(1);
        if (!pageBlocks[key].slug.startsWith("/")) pageBlocks[key].slug = `/${pageBlocks[key].slug}`;
        notionPageIdToSlugMapper[key] = pageBlocks[key].slug;
    }
    return {
        pageBlocks: pageBlocks,
        notionPageIdToSlugMapper: notionPageIdToSlugMapper,
        pageMap: pageMap
    };
}


export {$6592d0df9d7b02dd$export$2e2bcd8739ae039 as default};
